code/controllers/UserController.php

```diff
@@ -17,8 +17,9 @@ public function index()
     {
         if (!Member::currentUserID()) {
             return $this->redirect('/Security/login');
-        }
-        return $this->renderWith(array('UserController', 'Page'));
+        } else if (UserInvitation::create()->canCreate(Member::CurrentUser()) == false){
```

Comment:
Aah yes. Let's do this instead:
* Scrap the init idea and move the permission check `if(!Permission::check('ACCESS_USER_INVITATIONS')){   return Security::permissionFailure();}` to a method called permissionCheck().
* Introduce ` private static $allowed_actions = array('index' => '->permissionCheck');` in the controller to secure index by this permission check. This way the email viewing link should work.